### DISCUSSION TITLE
How can I test React Components with TurboRepo?

### DISCUSSION THREAD

**Discussion by MestreALMO** [vercel/turborepo#2623] [Help] [answered]
I'm having trouble intalling a testing library, I already have a application that I built from the begining with TurboRepo, but I can not figure out what I am missing to make tests work. I'm trying to test react components, but every time I have an error, usually it is something about the testing library not being able to read JSX elements, I feel like I am missing something in the configuration, but I can't figure out what.

I also tried to run "npx create-turbo@latest" with the objective of only making a new turborepo and create a new react component simple, Title, that has a function that returns a h1, and I tried to make a test file for this component and i get a error when importing this component, I've added the code that is said to add in the turborepo docs.

Usually I use create-react-app that already comes with a working testing library, but it is not included with turborepo.

Here are some librarys I tried to install in turborepo so I can make test working: jest react-scripts @testing-library/jest-dom @testing-library/react @testing-library/user-event @types/jest @types/node @types/react @types/react-dom ts-node ts-jest

**Comment by anthonyshew** [comment, marked as answer]
The `with-docker` example today has a Jest setup in it: https://github.com/vercel/turbo/tree/main/examples/with-docker
